EventFilter/Utilities/plugins/FRDStreamSource.cc

```diff
@@ -64,8 +64,10 @@ bool FRDStreamSource::setRunAndEventInfo(edm::EventID& id,
   }
 
   if (detectedFRDversion_ == 0) {
-    fin_.read((char*)&detectedFRDversion_, sizeof(uint32_t));
-    assert(detectedFRDversion_ > 0 && detectedFRDversion_ <= 5);
+    fin_.read((char*)&detectedFRDversion_, sizeof(uint16_t));
+    fin_.read((char*)&flags_, sizeof(uint16_t));
+    detectedFRDversion_ = detectedFRDversion_ & 0xffff;
```

Comment:
It seems to me this masking does not do anything.

Reply:
Not with the latest change. Corrected it now.